extensions/twitch.js

```diff
@@ -0,0 +1,297 @@
+      chat.style.width = args.width;
+      chat.style.height = args.height;
+      chat.style.border = "none";
+      chat.style.position = args.position;
+      chat.setAttribute("allowtransparency", "true");
+      chat.setAttribute("src", Link);
+      Scratch.renderer.addOverlay(chat, "manual");
+    }
+
+    closechat() {
+      if (chat == null) {
+        console.log("Can't close chat if chat wasn't opened.");
+      } else {
+        Scratch.renderer.removeOverlay(chat);
+        chat = null;
+      }
+    }
+
+    text(args) {
+      if (text == null) {
+        text = document.createElement("p");
+        text.style.font = args.font;
+        text.innerText = args.text;
+        text.width = "20%";
+        text.height = "10%";
+        text.style.border = "none";
+        text.style.position = "absolute";
+        text.style.bottom = 0;
+        if (args.bold) {
+          text.style.fontWeight = "bold";
+        }
+        if (args.italic) {
+          text.style.fontStyle = "italic";
+        }
+        text.style.fontSize = args.size;
+        text.style.fontFamily = args.font;
+        Scratch.renderer.addOverlay(text, "manual");
+      } else {
+        Scratch.renderer.removeOverlay(text);
+        text = document.createElement("p");
+        text.style.font = args.font;
+        text.innerText = args.text;
+        text.width = "20%";
+        text.height = "10%";
+        text.style.border = "none";
+        text.style.position = "absolute";
+        text.style.bottom = 0;
+        if (args.bold) {
+          text.style.fontWeight = "bold";
+        }
+        if (args.italic) {
+          text.style.fontStyle = "italic";
+        }
+        text.style.fontSize = args.size;
+        text.style.fontFamily = args.font;
+        Scratch.renderer.addOverlay(text, "manual");
+      }
+    }
+
+    image(args) {
+      if (image == null) {
+        image = document.createElement("img");
+        image.style.width = args.width;
+        image.style.height = args.height;
+        image.style.position = args.position;
+        if (args.bottom) {
+          image.style.bottom = 0;
+        } else {
+          if (args.top) {
+            image.style.top = 0;
+          }
+        }
+        image.setAttribute("src", args.image);
+        Scratch.renderer.addOverlay(image, "manual");
+      } else {
+        Scratch.renderer.removeOverlay(image);
+        image = document.createElement("img");
+        image.style.width = args.width;
+        image.style.height = args.height;
+        image.style.position = args.position;
+        if (args.bottom) {
+          image.style.bottom = 0;
+        } else {
+          if (args.top) {
+            image.style.top = 0;
+          }
+        }
+        image.setAttribute("src", args.image);
+        Scratch.renderer.addOverlay(image, "manual");
+      }
+    }
+
+    closetext() {
+      if (text == null) {
+        console.log("No text element has been created!");
```

Comment:
I don't think this console log is necessary.

Reply:
Turned into an alert.